pkg/providers/tinkerbell/hardware/config.go

```diff
@@ -89,6 +90,14 @@ func (hc *HardwareConfig) ValidateHardware(skipPowerActions bool, tinkHardwareMa
 			return fmt.Errorf("hardware id '%s' is not registered with tinkerbell stack", hw.Spec.ID)
 		}
 
+		hardwareInterface := tinkHardwareMap[hw.Spec.ID].GetNetwork().GetInterfaces()
+		if len(hardwareInterface) > 0 {
+			mac := hardwareInterface[0].GetDhcp()
```

Comment:
Maybe this is a broader discussion, but can we assume that the Mac of the first interface is what the workflow will map with? Not sure what happens if we specify more interfaces.